src/DynamoPackages/Package.cs

```diff
@@ -31,6 +30,41 @@ public string Name
         }
     }
 
+    public class PackageLoadState
+    {
+        public enum StateTypes
+        {
+            None, Loaded, Unloaded, Error
+        }
+
+        public enum ScheduledTypes
+        {
+            None, ScheduledForUnload, ScheduledForDeletion, ScheduledForLoad
+        }
+
+        private string errorMessage;
+        private ScheduledTypes scheduledState = ScheduledTypes.None;
+        private StateTypes state = StateTypes.None;// Default to None type.
+
+        internal void SetAsLoaded() { state = StateTypes.Loaded; errorMessage = ""; }
+        internal void SetAsError(string msg = "") { state = StateTypes.Error; errorMessage = msg; }
+        internal void SetAsUnloaded() { state = StateTypes.Unloaded; errorMessage = ""; }
+
+        internal void SetScheduledForDeletion() { scheduledState = ScheduledTypes.ScheduledForDeletion; }
+        internal void SetScheduledForUnload() { scheduledState = ScheduledTypes.ScheduledForUnload; }
+        internal void SetScheduledForLoad() { scheduledState = ScheduledTypes.ScheduledForLoad; }
+        internal void ResetScheduledState() { scheduledState = ScheduledTypes.None; }
```

Comment:
I went with the option of using explicit setter methods (ex. SetAsUnloaded) because it makes it easier to use the Error state + error message.